### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by jake-mcg**

### CODE REVIEW

File: Inheritance/src/Inheritance/ActorExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    {
+        public static string SayLine(this Actor actor) =>
+            actor switch
+        {
+            Sheldon sheldon => $"{actor.GetType().Name}: { sheldon.Script}",
+            Penny penny => $"{actor.GetType().Name}: { penny.Script}",
+            Raj raj when raj.AreWomenPresent => $"{actor.GetType().Name}: {raj.Script}",
+            Raj raj when !raj.AreWomenPresent => $"{actor.GetType().Name}: {raj.Script}",
+            { } => string.Empty
+        };
```
Comment (pnotequalnp) [inline Inheritance/src/Inheritance/ActorExtensions.cs]: This will fail when actor is null.

File: Inheritance/src/Inheritance/ActorExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Inheritance
+{
+    public static class ActorExtensions
+    {
+        public static string SayLine(this Actor actor) =>
```
Comment (pnotequalnp) [inline Inheritance/src/Inheritance/ActorExtensions.cs]: The assignment specs wanted this method called "Speak".

File: Inheritance/test/Inheritance.Tests/FoodTests.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+                    // Assert
+                    using (var reader = new StreamReader(stream))
+                    {
+                        var lineWritten = reader.ReadLine();
+                        Assert.AreEqual("<Test Upc> - <Test Brand>", lineWritten);
+                    }
+                }
+            }
+        }
```
Comment (pnotequalnp) [inline Inheritance/test/Inheritance.Tests/FoodTests.cs]: This tests both Food and Printer at the same time, better to isolate them and test each individually.

File: Inheritance/test/Inheritance.Tests/TelevisionTests.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+                    // Assert
+                    using (var reader = new StreamReader(stream))
+                    {
+                        var lineWritten = reader.ReadLine();
+                        Assert.AreEqual("<Test Manufacturer> - <Test Size>", lineWritten);
+                    }
+                }
+            }
+        }
```
Comment (pnotequalnp) [inline Inheritance/test/Inheritance.Tests/TelevisionTests.cs]: Same as with FoodTests

File: Inheritance/src/Inheritance/ActorExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+namespace Inheritance
+{
+    public static class ActorExtensions
+    {
+        public static string Speak(this Actor actor) =>
+            actor switch
+        {
+            Sheldon sheldon                     => $"{actor.GetType().Name}: { sheldon.Script}",
+            Penny penny                         => $"{actor.GetType().Name}: { penny.Script}",
+            Raj raj when raj.AreWomenPresent    => $"{actor.GetType().Name}: {raj.Script}",
```
Comment (breaman) [inline Inheritance/src/Inheritance/ActorExtensions.cs]: What is the difference between the two calls for Raj? They both call the same method, so you should have had them call different methods and then not have the if check in the Script method.

**Pull request closed without merging**